frontend/src/container/NewWidget/LeftContainer/WidgetGraph/WidgetGraphs.tsx

```diff
@@ -64,14 +70,47 @@ function WidgetGraph({
 		(start: number, end: number): void => {
 			const startTimestamp = Math.trunc(start);
 			const endTimestamp = Math.trunc(end);
-
 			if (startTimestamp !== endTimestamp) {
 				dispatch(UpdateTimeInterval('custom', [startTimestamp, endTimestamp]));
 			}
+
+			const { maxTime, minTime } = GetMinMax('custom', [
+				startTimestamp,
+				endTimestamp,
+			]);
+
+			params.set(QueryParams.startTime, minTime.toString());
+			params.set(QueryParams.endTime, maxTime.toString());
+			const generatedUrl = `${location.pathname}?${params.toString()}`;
+			history.push(generatedUrl);
 		},
-		[dispatch],
+		[dispatch, location.pathname, params],
 	);
 
+	const handleBackNavigation = (): void => {
```

Comment:
This function and useEffect are identical as in `frontend/src/container/GridCardLayout/GridCard/index.tsx`, 
Can we create a custom hook and use it at both places?

Reply:
yes have already taken note of it , will do it separately

Reply:
For those things which we are going to take in the separate PR, we should add TODO as a note above the function, so that it doesn't miss. 

cc: @YounixM